consensus/src/metrics_safety_rules.rs

```diff
@@ -125,3 +148,121 @@ impl TSafetyRules for MetricsSafetyRules {
         })
     }
 }
+
+pub struct MockSafetyRules {
+    // number of initialize() calls
+    init_calls: i32,
+
+    // max initialize() calls to complete perform_initialize()
+    max_init_calls: i32,
+}
+
+impl MockSafetyRules {
+    #[allow(dead_code)]
+    pub fn new(init_calls: i32, max_init_calls: i32) -> Self {
+        Self {
+            init_calls,
+            max_init_calls,
+        }
+    }
+}
+
+impl TSafetyRules for MockSafetyRules {
+    fn consensus_state(&mut self) -> Result<ConsensusState, Error> {
+        let safety_data = SafetyData::new(
+            0,    /* epoch */
+            0,    /*last_voted_round */
+            0,    /* preferred_round*/
+            0,    /* one_chain_round */
+            None, /* last_vote */
+        );
+        let waypoint = Waypoint::new_any(&LedgerInfo::new(
+            BlockInfo::new(
+                0,                 /* epoch */
+                0,                 /* round */
+                HashValue::zero(), /* id */
+                HashValue::zero(), /* executed_state_id */
+                0,                 /* version */
+                0,                 /* timestamp_usecs */
+                None,              /* next_epoch_state */
+            ),
+            HashValue::zero(), /* consensus_data_hash */
+        ));
+        Ok(ConsensusState::new(
+            safety_data,
+            waypoint,
+            true, /* in_validator_set */
+        ))
+    }
+
+    fn initialize(&mut self, _: &EpochChangeProof) -> Result<(), Error> {
+        self.init_calls += 1;
+        // Emulate scenario where max_init_calls of calls to initialize()
```

Comment:
nit: you could probably also remove this comment. I think the code is pretty clear 😄 Or, at the very least, shorten it, e.g., `Initialize fails the first "max_init_calls" times, after which it will succeed.`

Reply:
Removed the comment.